### PR TITLE
Update canister ID resolution based on custom domain changes

### PR DESCRIPTION
**Pull request by r-birkner**

### CODE REVIEW

File: spec/index.adoc [hunk 1]
```diff
@@ -2165,22 +2165,27 @@ NOTE: Canisters not using these features can completely leave out the `streaming
...
+| `identity.ic0.app`   | `rdmx6-jaaaa-aaaaa-aaadq-cai`
+| `nns.ic0.app`        | `qoctq-giaaa-aaaaa-aaaea-cai`
+| `dscvr.one`          | `h5aet-waaaa-aaaab-qaamq-cai`
+| `dscvr.ic0.app`      | `h5aet-waaaa-aaaab-qaamq-cai`
+| `personhood.ic0.app` | `g3wsl-eqaaa-aaaan-aaaaa-cai`
+|=====================================================
+
+2. Check whether the hostname is a _raw_ hostname (e.g., `<name>.raw.ic0.app`). If it is the case, fail and handle the request as a Web2 request, otherwise continue.
+
+3. Check whether the canister ID is embedded in the hostname by splitting the hostname and finding the first occurrence of a valid canister ID from the right. If there is a canister ID embedded in the hostname, use it.
```
Comment (keplervital) [inline spec/index.adoc:2182]: We also take the `topdomains` to determine the gateway to call.  

`rdmx6-jaaaa-aaaaa-aaadq-cai.ic0.app` will extract the principal and use `ic0.app` as the api gateway.

File: spec/index.adoc [hunk 2]
```diff
@@ -2165,22 +2165,27 @@ NOTE: Canisters not using these features can completely leave out the `streaming
...
+
+4. Check whether there is a `canisterID` parameter in the URL.
 
-3. Else, if `<name>` is a valid textual encoding of a principal, use that principal as the canister id.
+5. Make a `HEAD` request to the hostname. If the response contains an `x-ic-canister-id`- and `x-ic-gateway`-header, use the canister ID specified in the header.
```
Comment (keplervital) [inline spec/index.adoc:2186]:
```suggestion
5. Make a `HEAD` request to the hostname. If the response contains an `x-ic-canister-id` and `x-ic-gateway` header, use the canister ID specified in the header.
```

**Review by keplervital — APPROVED** [review]

**Review by Dfinity-Bjoern — APPROVED** [review]
Thanks @r-birkner!

File: spec/index.adoc [hunk 3]
```diff
@@ -2165,22 +2165,29 @@ NOTE: Canisters not using these features can completely leave out the `streaming
...
+| `dscvr.one`          | `h5aet-waaaa-aaaab-qaamq-cai`
+| `dscvr.ic0.app`      | `h5aet-waaaa-aaaab-qaamq-cai`
+| `personhood.ic0.app` | `g3wsl-eqaaa-aaaan-aaaaa-cai`
+|=====================================================
+
+2. Check whether the hostname is a _raw_ hostname (e.g., `<name>.raw.ic0.app`). If it is the case, fail and handle the request as a Web2 request, otherwise continue.
+
+3. Check whether the canister ID is embedded in the hostname by splitting the hostname and finding the first occurrence of a valid canister ID from the right. The part of the hostname to the right of the canister ID, is the gateway (e.g., `ic0.app` is the gateway for `rdmx6-jaaaa-aaaaa-aaadq-cai.ic0.app`). If there is a canister ID embedded in the hostname, use it.
+
+4. Check whether the canister is hosted on the IC using a custom domain. There are two options:
```
Comment (Dfinity-Bjoern) [inline spec/index.adoc:2184]:
```suggestion
4. Check whether the canister is hosted on the IC using a custom domain. The client-side implementation can choose one of the following two options:
```

**Pull request merged**